dev-util/dub/dub-1.36.0-r1.ebuild

```diff
@@ -0,0 +1,171 @@
+
+	# Generate man pages. Rebuids dub so put it behind a USE flag.
+	if use doc; then
+		einfo "Generating man pages"
+		# You're supposed to be able to do ./bin/dub scrips/man/gen_man.d
+		# but it gives linking errors with gdc.
+
+		# $imports is set up above.
+		versions=DubUseCurl
+		dlang_compile_bin scripts/man/gen_man{,.d} \
+						  $(sed '/^source\/app.d$/d' build-files.txt)
+		./scripts/man/gen_man || die "Could not generate man pages"
+	fi
+}
+
+src_test() {
+	# Setup the environment for the tests.
+	local -x DUB="${S}/bin/dub"
+	local -x DUB_HOME="${T}/dub-home" # where to put artifacts
+
+	# Note, disabling tests is possible yet very hard. You have to
+	# create a bash variable containing a regex (to be used in =~) that
+	# matches all the tests that you want *to* run. It's probably easier
+	# to delete the subdirectory under ${S}/test.
+
+	# Tries to connect to github.com and fails due to the network sandbox
+	rm -rf "${S}/test/git-dependency" || die
+	# Doesn't work on non amd64/x86
+	if [[ ${ARCH} == arm64 ]]; then
+		rm -rf test/issue1447-build-settings-vars || die
+	fi
+
+	local dropImportCTest
+	# We have an importC test and not all compilers pass it properly.
+	# gdc doesn't support #include's in its importC implementation yet.
+	# Only check == 13 since 12 is skipped by the script.
+	[[ ${EDC} == gdc* ]] && [[ $(dlang_get_be_version) == 13 ]] && dropImportCTest=1
+	# Nor does <=ldc2-1.32.
+	[[ ${EDC} == ldc* ]] && $(ver_test $(dlang_get_be_version) -le 1.32) && dropImportCTest=1
+	# dmd can do #include's but there are some other errors about
+	# __float128 in <=dmd-2.102 on x86.
+	[[ ${EDC} == dmd* ]] && $(ver_test $(dlang_get_be_version) -le 2.102) \
+		&& [[ ${ARCH} == x86 ]] && dropImportCTest=1
+	if [[ ${dropImportCTest} ]]; then
+		rm -rf "${S}/test/use-c-sources" || die
+	fi
+
+	# Put the configuration file relative to the dub binary:
+	# <dub-bin-dir>/../etc/dub/settings.json as per
+	# https://dub.pm/dub-reference/settings so that it's picked up
+	# automatically.
+	mkdir -p "${S}/bin/../etc/dub" || die
+	cp "${T}/settings.json" "${S}/bin/../etc/dub/" \
+		|| die "Could not copy dub configuration file"
+
+	# See https://bugs.gentoo.org/921581 we have to remove -op (preserve
+	# source path for output files) from the flags lest the sandbox
+	# trips us up. This shouldn't be a problem anymore with dlang-single.
+	local newDflags="${DCFLAGS//--op/}"
+	newDflags="${newDflags//-op/}"
+
+	# Append -Wno-error or equivalent
+	newDflags+=" $(get_wno_error_flag)"
+
+	DFLAGS="${newDflags}" FRONTEND="$(dlang_get_fe_version)" test/run-unittest.sh  \
+		|| die "Tests failed"
+}
+
+src_install() {
+	dobin bin/dub
+	dodoc README.md
+
+	# Make sure there are no man files in any other section.
+	use doc && doman scripts/man/*.1
+
+	newbashcomp scripts/bash-completion/${PN}.bash ${PN}
+	dozshcomp scripts/zsh-completion/_${PN}
+	dofishcomp scripts/fish-completion/${PN}.fish
+}
+
+get_wno_error_flag() {
```

Comment:
Again feels like it belongs in an eclass?

Reply:
I would say that this issue is mostly relevant when building with `dub` since that one adds -Werror by default so I would say that it would best go into a dub.eclass. Still, I think there are plenty of packages which hardcode flags like -Werror -O -inline and similar so I also see it possible to add this to the eclass. My only argument would be that the function is so simple that it would be better to make _dlang_echo_implementation_string public and have the ebuilds call it with `-wi -Wno-error --wi` but I admit that my upbringing in programming makes me overvalue such code-golf solution.

In the end I think I'll follow your advice and add it to dlang-utils.